### ISSUE TITLE
prtrain model

### ISSUE THREAD

**Issue by longzeyilang** [leon-liangwu/py-caffe-yolo#1]
It uses pretrain model in train_tiny.sh, where can I download the darknet_v1.9.caffemodel?

**Comment by leon-liangwu** [comment]
@longzeyilang 
I do not have the pre-trained right now, because it has been a little long time since this repo developed.
I suggest you use my new repo  [MaskYolo_Caffe](https://github.com/leon-liangwu/MaskYolo_Caffe.git), which contains a lot of details on how to run the demos.
You can train your own models for object detection.

**Comment by longzeyilang** [comment]
请问怎么修改config.py的配置参数？

**Comment by leon-liangwu** [comment]
@longzeyilang 你说的config.py 是什么caffe里是net.prototxt 和 solver.prototxt

**Comment by longzeyilang** [comment]
能否对boxdatalayer.py的输出参数的格式说下？

**Comment by longzeyilang** [comment]
然后regionlosslayer中self.data_seen < 12800，为什么要这么做？

**Comment by longzeyilang** [comment]
self.anchors调整的依据什么？

**Comment by leon-liangwu** [comment]
@longzeyilang boxdatalayer.py 的输出就是图像和box的groundtruth，这些在网络和代码里可以看到。
self.data_seen < 12800这个在原作者的代码里也是有的，原始是用来在训练初期得到一个比较好的初始化的参数。
self.anchors 可以根据gt box进行kmeans，我这里有写好的脚本，可以发你参考。

**Comment by longzeyilang** [comment]
输入不是gt的真实坐标，而是box:[label,xcenter_scale,ycenter_scale,width_scale,height_scale]，xcenter_scale=xcenter/w   ycenter_scale=ycenter/h  width_scale=roi_width/w   height_scale=roi_height/h

**Comment by longzeyilang** [comment]
self.anchors脚本能否上传下？多谢

**Comment by longzeyilang** [comment]
def convert(self,size,box):
    dw=1./size[0]
    dh=1./size[1]
    x=(box[0]+box[2])/2.0-1
    y=(box[1]+box[3])/2.0-1
    w=box[2]-box[0]
    h=box[3]-box[1]
    x=x*dw
    w=w*dw
    y=y*dh
    h=h*dh
    return (x,y,w,h)

**Comment by leon-liangwu** [comment]
@longzeyilang 你好 gtbox是经过normized，这样图像可以任意缩放。
我上传了kmeans_anchor.py 在 [MaskYolo_Caffe](https://github.com/leon-liangwu/MaskYolo_Caffe.git)这个repo的scripts里，请查看。

**Comment by longzeyilang** [comment]
还有一个问题regionlosslayer中：      
        i = int(truth[0] * self.w)   #位置
        j = int(truth[1] * self.h)
        truth_shift = truth.copy()
        truth_shift[0] = 0    #这里为什么要置0
        truth_shift[1] = 0

**Comment by longzeyilang** [comment]
kmeans_anchor.py的数据输入每条记录的格式是怎么样？

**Comment by leon-liangwu** [comment]
@truth_shift[0] = 0 truth_shift[1] = 0 是为了不考虑中心点的位置跟anchor做iou计算。这一段代码遍历每一个gt，并且和当前位置上的所有anchor做iou，选出w,h最合适的anchor。
kmeans_anchor.py的数据输入每条记录的格式和原作者的一样。输入是一个txt文件，每行是图像的位置。image的path替换某些字符串找到label的path。
```
img_list = open('/path/to/img_list.txt')
net_w, net_h = 608, 608  # input w & h
num_cls = 3 # number of classes
```

**Comment by longzeyilang** [comment]
不是这个意思，是否txt文本中每行[label,xcenter,ycenter,width,height]，这个是normalize后gt

**Comment by leon-liangwu** [comment]
@longzeyilang 对 是normalized之后的gt 和darknet的label一样。

**Comment by longzeyilang** [comment]
能发一条kmeans_anchor.py的数据输入的记录？我看我这边是否写错了

**Comment by leon-liangwu** [comment]
@longzeyilang  是这个吗？ 
```
1 0.766 0.605333333333 0.196 0.368
```

**Comment by longzeyilang** [comment]
kmeans_anchor.py中得到final_centers，放在config.py中anchors需要乘以net_w？这边出现regionlosslayer.py中get_region_box的x[2]=250，导致数据过大，抛出错误

**Comment by leon-liangwu** [comment]
@longzeyilang 
yolo v2 使用kmeans_anchor.py的时候，`net_w, net_h = xxx, xxx ` 设置成输出feature map的长宽。yolo v3设置成输入的长宽。这是要注意的

**Comment by longzeyilang** [comment]
给点建议，这些注意方面能否添加下？感觉资料缺少很多

**Comment by leon-liangwu** [comment]
@longzeyilang 
这些是和darknet原框架一样的实现，所以它那边要注意的这边也同样要注意。
我建议你尝试一下，MaskYolo_Caffe 的demo，那边写的比较清楚。
